crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -43,7 +43,7 @@ const TRANSFER_PORT_ID: &str = "transfer";
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ChainId {
     id: String,
-    revision_number: u64,
+    revision_number: Option<u64>,
```

Comment:
From what I see, the primary focus here is on making `ChainId`'s specific format (`{chain_name}-{revision_number}`) optional, rather than the revision number. Thus, even if a `ChainId` doesn't adhere to the current format, it should still have a default revision number (0). Specifically the `Option` here serves the purpose of specifying whether the ChainId complies with the specified format. 
Note that since the `id` field fully represents the `ChainId`, it's possible to detect whether a `ChainId` includes a revision number through the 'id' itself. Therefore, I believe our solution should involve introducing a default revision number of zero when parsing the ChainId string, in cases where the `id` does not conform to the expected format and can keep the data structure as is.

Reply:
I wanted to make a distinction between _zero revision number_ and _no revision number_. I agree that `has_revision_number` can reuse `split_chain_id` - but I wanted to call it once when a chain ID is created/parsed and save the information as `Option<u64>`.